tests/spark/test_spark_connect_model_export.py

```diff
@@ -0,0 +1,452 @@
+    artifact_repo.log_artifacts(model_path, artifact_path=artifact_path)
+
+    model_uri = artifact_root + "/" + artifact_path
+    reloaded_model = sparkm.load_model(model_uri=model_uri)
+    preds_df = reloaded_model.transform(spark_model_iris.pandas_df.copy(deep=False))
+    pd.testing.assert_frame_equal(spark_model_iris.predictions, preds_df, check_dtype=False)
+
+
+def test_log_model_calls_register_model(spark_model_iris):
+    artifact_path = "model"
+    register_model_patch = mock.patch("mlflow.tracking._model_registry.fluent._register_model")
+    with mlflow.start_run(), register_model_patch, TempDir(chdr=True, remove_on_exit=True) as tmp:
+        sparkm.log_model(
+            spark_model_iris.model,
+            artifact_path=artifact_path,
+            registered_model_name="AdsModel1",
+        )
+        model_uri = f"runs:/{mlflow.active_run().info.run_id}/{artifact_path}"
+        assert_register_model_called_with_local_model_path(
+            register_model_mock=mlflow.tracking._model_registry.fluent._register_model,
+            model_uri=model_uri,
+            registered_model_name="AdsModel1",
+        )
+
+
+def test_log_model_no_registered_model_name(spark_model_iris):
+    artifact_path = "model"
+    register_model_patch = mock.patch("mlflow.tracking._model_registry.fluent._register_model")
+    with mlflow.start_run(), register_model_patch, TempDir(chdr=True, remove_on_exit=True) as tmp:
+        sparkm.log_model(
+            spark_model_iris.model, artifact_path=artifact_path,
+        )
+        mlflow.tracking._model_registry.fluent._register_model.assert_not_called()
+
+
+def test_model_save_persists_specified_conda_env_in_mlflow_model_directory(
+    spark_model_iris, model_path, spark_connect_model_custom_env
+):
+    sparkm.save_model(
+        spark_model_iris.model,
+        path=model_path,
+        conda_env=spark_connect_model_custom_env
+    )
+
+    pyfunc_conf = _get_flavor_configuration(model_path=model_path, flavor_name=pyfunc.FLAVOR_NAME)
+    saved_conda_env_path = os.path.join(model_path, pyfunc_conf[pyfunc.ENV]["conda"])
+    assert os.path.exists(saved_conda_env_path)
+    assert saved_conda_env_path != spark_connect_model_custom_env
+
+    with open(spark_connect_model_custom_env) as f:
+        custom_env_parsed = yaml.safe_load(f)
+    with open(saved_conda_env_path) as f:
+        saved_conda_env_parsed = yaml.safe_load(f)
+    assert saved_conda_env_parsed == custom_env_parsed
+
+
+def test_model_save_persists_requirements_in_mlflow_model_directory(
+    spark_model_iris, model_path, spark_connect_model_custom_env
+):
+    sparkm.save_model(
+        spark_model_iris.model, path=model_path,
+        conda_env=spark_connect_model_custom_env
+    )
+
+    saved_pip_req_path = os.path.join(model_path, "requirements.txt")
+    _compare_conda_env_requirements(spark_connect_model_custom_env, saved_pip_req_path)
+
+
+def test_log_model_with_pip_requirements(spark_model_iris, tmp_path):
```

Comment:
Do we need these tests? `tests/spark/test_spark_model_export.py` already has them.

Reply:
The same comment on other tests.

Reply:
@WeichenXu123

> can we ensure spark connect model compatible with these arguments ?

We're already testing model registration or pip requirement arguments in `tests/spark/test_spark_model_export.py`. This PR just updates how we serialize the model, and doesn't touch other parts, right?

Reply:
When we have a function like this, do we need to test `save_pip_requirements` for both 
`SparkConnectModel` and non-`SparkConnectModel` model?

```python
def log_model(model, pip_requirements):
    if isinstance(model, SparkConnectModel):
        save_spark_connect_model(model)
    else:
        save_model(model)

    save_pip_requirements(pip_requirements)
```

Reply:
I'm not saying to remove all the tests.